src/scripts/modules/gradebook.js

```diff
@@ -257,17 +260,71 @@
         return students;
     }
 
+    function getPathContent(path) {
+      const pathURI = id => `https://online.theironyard.com/admin/paths/${ id }`;
+
+      return new Promise( ( res ) => {
```

Comment:
This is a bit problematic.. If the `$.get()` fails then we'll never see that since you never reject this Promise. I think we could just `return` on line 271 (the `$.get().then()` call) but we'd also need to `return` the `path` variable on line 275 (instead of execute the resolution). For this all to work, we'd also need to return the `new Promise()`, but only in the positive `if` block (line 269, essentially).

The alternative is to add a `.fail()` to the `$.get()` call and execute the `Promise` rejection callback.